src/AutoRest.CSharp/LowLevel/Decorator/MethodAmbiguityResolver.cs

```diff
@@ -0,0 +1,169 @@
+        private static LowLevelClientMethod ResolveClientMethod(LowLevelClientMethod clientMethod, IReadOnlyList<IMethodSymbol> clientMethodSourceModels)
+        {
+            bool ambiguityFound = false;
+            foreach (var clientMethodSourceModel in clientMethodSourceModels)
+            {
+                if (IsAmbiguousMethod(clientMethod, clientMethodSourceModel))
+                {
+                    // if two methods are identical, that means there should be some customization codes and we leave them to users to resolve
+                    if (IsSameMethod(clientMethod, clientMethodSourceModel))
+                    {
+                        return clientMethod;
+                    }
+                    ambiguityFound = true;
+                }
+            };
+
+            if (ambiguityFound)
+            {
+                return clientMethod.CloneWithAllParametersRequired();
+            }
+            return clientMethod;
+        }
+
+        /// <summary>
+        /// Check if a client method is ambiguous with the method in source model.
+        /// Two methods are considered as ambiguous when:
+        /// - both has the same number of required parameters
+        /// - for each required parameter, either they should be the same type in source model.
+        /// </summary>
+        /// <param name="clientMethod"></param>
+        /// <param name="methodSymbol"></param>
+        /// <returns></returns>
+        private static bool IsAmbiguousMethod(LowLevelClientMethod clientMethod, IMethodSymbol methodSymbol)
+        {
+            var clientMethodRequiredParameters = clientMethod.Signature.Parameters.Where(p => !p.IsOptionalInSignature).ToList();
+            var requiredParameterSymbols = methodSymbol.Parameters.Where(p => !p.HasExplicitDefaultValue).ToList();
+            if (clientMethodRequiredParameters.Count != requiredParameterSymbols.Count)
+            {
+                return false;
+            }
+
+            for (int i = 0; i < clientMethodRequiredParameters.Count; i++)
+            {
+                if (!IsSameParameter(clientMethodRequiredParameters[i], requiredParameterSymbols[i]))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Check if a clientMethod has the same paramters as a method from source model has.
+        /// Both methods should:
+        /// - have the same number of parameters
+        /// - each parameter has the same type
+        /// - each parameter has the same default value (e.g. optional or required)
+        /// </summary>
+        /// <param name="clientMethod"></param>
+        /// <param name="clientMethodSourceModel"></param>
+        /// <returns></returns>
+        private static bool IsSameMethod(LowLevelClientMethod clientMethod, IMethodSymbol clientMethodSourceModel)
```

Comment:
IsMethodSignatureSame?

Reply:
That's a better name. I'll make the change. Thanks.